Modules/Quotas/src/Quotas.Domain/Aggregates/Identities/Identity.cs

```diff
@@ -23,42 +31,67 @@ public void AssignTierQuotaFromDefinition(TierQuotaDefinition definition)
         _tierQuotas.Add(tierQuota);
     }
 
-    public async Task UpdateMetrics(IEnumerable<string> metrics, IMetricCalculatorFactory factory, CancellationToken cancellationToken)
+    private void UpdateMetricStatus(MetricKey metricKey, IMetricCalculator metricCalculator)
     {
-        foreach (var metric in metrics)
+        var metricStatus = _metricStatuses.Where(m => m.MetricKey == metricKey).FirstOrDefault();
+        if(metricStatus != null)
         {
-            var metricCalculator = factory.CreateFor(metric);
-            await UpdateMetricAsync(metric, metricCalculator, cancellationToken);
+            var quotasOfMetric = GetAppliedQuotasForMetric(metricKey);
+            metricStatus.Update(quotasOfMetric.Max(q => q.IsExhaustedUntil));
         }
-        return;
     }
 
-    private async Task UpdateMetricAsync(string metric, IMetricCalculator metricCalculator, CancellationToken cancellationToken)
+    public void UpdateAllMetricStatuses()
```

Comment:
We're currently using it for tests, and I figure we may need it in the foreseable future.

As for the way I'm using it, it's a mere matter of simplicity. The way I understand the `_metricStatuses` attribute, it should actually be a `Map<MetricKey, MetricStatus>` or `Map<MetricKey, DateTime? (IsExhaustedUntil)>`. That would certainly simplify this.

Reply:
> We're currently using it for tests, and I figure we may need it in the foreseable future.

We will never update just MetricStatuses. It will only happen via `UpdateMetric` or `UpdateAllMetrics`. Plus: having a public method only for a test is a contradiction by itself. :) Because why would you test a method that is not used anywhere in the production code?

> As for the way I'm using it, it's a mere matter of simplicity. The way I understand the _metricStatuses attribute, it should actually be a Map<MetricKey, MetricStatus> or Map<MetricKey, DateTime? (IsExhaustedUntil)>. That would certainly simplify this.

I agree that an Map would be a better fit, but I'm not sure whether EF Core can handle this. But since the only class that accesses the _metricKeys is the `Identity` class, we can always check if the `MetricKey` is already present. But I don't see how a Map would make any difference in the process of updating the MetricStatuses

Reply:
By removing that method, we've destroyed any way to refresh the MetricStatuses for a given Identity when there's no change to quota's usage.

I've been going round and round about this and I just noticed that you mentioned the `UpdateAllMetrics` method. We do not have such a thing, though it does sound terribly similar to the method I just removed. 

So, I suppose the question is should we have a way to refresh the MetricStatuses for a given Identity (a smart thing to do regularly because quotas can become available again), even if we're not using it in our code just yet (but it is needed for [this](https://github.com/nmshd/backbone/pull/160#discussion_r1236636894))?

Reply:
> even if we're not using it

Never write code that you don't use, even when you think that you will use it some day in the future. :) Such code needs to be maintained, and who knows, maybe we some day we decide that we don't even need it at all. Then all the effort for implementation and maintenance was in vain.

Reply:
Premature optimization is the root of all evil, yes. Can I then ditch the test I linked?

Reply:
That's not what I meant. :) I aimed at the UpdateAllMetricStatuses method. 
But still: when I think about it, you can ignore the comment you linked, because an IsExhaustedUntil date in the past is synonymous with a null value. Let's save the database capacity that would be needed for the update.